src/plot_utils.py

```diff
@@ -0,0 +1,181 @@
+import matplotlib.pyplot as plt 
+import numpy as np
+import os
+import pandas as pd
+import torch as th
+import matplotlib.pyplot as plt
+
+def plot_losses(losses, val_losses, model_name):
+    epochs = np.arange(len(losses))
+
+    plt.subplot(1, 2, 1)
+    plt.title(f"{model_name} - Power Flow Training Learning Curve")
+    plt.plot(epochs, losses, label="Training Loss")
+    plt.legend()
+    plt.xlabel("Epochs")
+    plt.ylabel("MSE")
+
+    plt.subplot(1, 2, 2)
+    plt.title(f"{model_name} - Power Flow Validation Learning Curve")
+    plt.plot(epochs, val_losses, label="Validation Loss")
+    plt.legend()
+    plt.xlabel("Epochs")
+    plt.ylabel("MSE")
+
+    plt.tight_layout()
+    plt.show()
+
+
+def distance_plot(model, batch):
+    """
+    Plots the error with distance from the generator for a given model and batch.
+
+    Args:
+        model (torch.nn.Module): The model to use for generating predictions.
+        batch (torch_geometric.data.Batch): The batch of data to use for plotting.
+
+    Returns:
+        None
+    """
+    out = model(batch)
+    distance_loss,len = get_distance_loss(out,batch.y,batch)
+    plt.bar(list(range(0,len)), distance_loss, color ='maroon')
+    plt.title("Error with distance from the generator")
+    plt.ylabel("Error")
+    plt.xticks(range(0,len))
+    plt.xlabel("Nodes away from the generator the node was located")
+    plt.show()
+    plt.savefig("distance_plot.png")
```

Comment:
@AlexDeLos save fig needs to be before show() otherwise its gonna save a blank image, also my changes have been deleted, I had added a better naming for the png file please check my previous commits